content/post/postgres/rfc-extension-packaging-lookup.md

```diff
@@ -0,0 +1,496 @@
+directory into which to install an extension, and will default to
+`--extdir-site`. It can be set to the values `$extdir_site`, `$extdir_vendor`,
+or `$extdir_core`, or to any literal path.
+
+The `$EXTENSION` variable will be changed to allow only one extension name. If
+it's set, the installation behavior will be changed for the following
+variables:
+
+*   `EXTENSION`: Creates `$EXTDIR/$EXTENSION`, installs
+    `$EXTDIR/$EXTENSION/$EXTENSION.control`
+*   `MODULES` and `MODULE_big`: Installed into `$EXTDIR/$EXTENSION/lib`
+*   `MODULEDIR`: Removed
+*   `DATA`: Installed into `$EXTDIR/$EXTENSION/share`
+*   `DATA_built`: Installed into `$EXTDIR/$EXTENSION/share`
+*   `DATA_TSEARCH`: Installed into `$EXTDIR/$EXTENSION/share/tsearch_data`
+*   `DOCS`: Installed into `$EXTDIR/$EXTENSION/doc`
+*   `PROGRAM`, `SCRIPTS` and `SCRIPTS_built`: Installed into
+    `$EXTDIR/$EXTENSION/bin`
+
+Another new variable, `LINKBINS`, will default to true and symlink
+`$EXTDIR/$EXTENSION/bin` files in `pg_config --bindir`. Installers can set it
+to false to skip the symlinking, e.g., for immutable Postgres installs.
+
+> [!NOTE] External projects that install extensions without using PGXS, like
+> [pgrx], must also be updated to either follow the same pattern or to
+> delegate installation to [PGXS].
+
+### MODULE_PATHNAME
+
+Update the installer to replace `MODULE_PATHNAME` in SQL scripts with the new
+install path for shared modules, `$EXTDIR/$EXTENSION/lib`.
+
+### Control File
+
+The `directory` and `module_pathname` control file variables will be
+deprecated and ignored.
+
+## Use Cases
+
+Here’s how the proposed file layout and `extension_path` GUC address the [use
+cases](#the-problem) that inspired this RFC.
+
+### Packager Testing
+
+A packager who wants to run tests without modifying a PostgreSQL install would
+follow these steps:
+
+*   Set the `extension_path` GUC to search the site extension directory under
+    the packaging install. Something like
+    `$RPM_BUILD_ROOT/$(pg_config --extdir-vendor)`
+*   Install the extension into that directory:
+    `make install EXTDIR=$RPM_BUILD_ROOT`
+*   Run `make installcheck`
+
+This will allow PostgreSQL to find and load the extension during the tests.
+The Postgres installation will not have been modified; only the
+`extension_path` will have changed.
+
+### Docker/Kubernetes
+
+To allow extensions to be added to a Docker container and to persist beyond
+its lifetime, one or more [volumes] could be used. A couple of options:
+
+*   Mount the `--extdir-site` and/or `--extdir-vendor` directories as
+    persistent volumes (or one volume and a subdirectory for each). Then any
+    extensions installed into them will persist. Files for any one extension
+    will live on a single volume. If a new container spins up, as long as it
+    uses the same persistent volume(s), can access the same extensions.
+
+*   Create separate images for each extension, and then "install" them by
```

Comment:
This feature is still marked as `alpha`. It should be mentioned.

Reply:
Added in 15c4952.